javaslang/src/main/java/javaslang/collection/Stream.java

```diff
@@ -980,6 +980,17 @@ default int length() {
     }
 
     @Override
+    default Stream<T> leftPadTo(int length, T element) {
+        if (length <= 1) {
```

Comment:
+1 for using `length <= thisLen` and adding a javadoc comment to Seq.java#leftPadTo, sth. like:

```java
/**
 ...
 * <p>
 * Please note that lazily-evaluated Seq implementations need to process all elements in order to gather the overall length.
 ...
 */
```

---

A second thing: We should also unify the if-clauses in all methods by using an else branch instead of an if-guard and evaluate the length only once by introducing a local variable.

```java
final int thisLength = length();
if (length <= thisLength) {
    ...
} else {
    ...
}
```

(In all classes)

Many thanks!